domapping/mapping.py

```diff
@@ -179,14 +179,43 @@ def _gen_type_properties(json_schema, path, resolver, config, es_mapping):
         resolver.push_scope(json_schema.get('id'))
 
     # resolve reference if there are any
+
+    def dict_search_and_retrieve(d, key=None):
```

Comment:
If I understand well you are trying to recursively find all values under the `key`. Consider yielding the values so the `any` function can evaluate to `True` early.

Try to find some descriptive name for the function (e.g. `iter_all_values_from_dict(data, key)`) and add some tests.